### PR TITLE
Build Images using Docker

### PR DESCRIPTION
**Pull request by hkfuertes**
Simple scripts to build the images using docker instead of vagrant:
```shell
git clone --recurse-submodules <repo>
docker-compose run build_XXXX # See docker-compose.yml for available options.
```

> This was useful to me, that's why I share :man_shrugging:

### CODE REVIEW

**Comment by nisargjhaveri** [comment]
@hkfuertes any reason to not merge this? Sorry for the delay, but this looks useful. I am just waiting to try this out as I'm not very familiar with docker yet.

**Comment by hkfuertes** [comment]
> @hkfuertes any reason to not merge this? Sorry for the delay, but this looks useful. I am just waiting to try this out as I'm not very familiar with docker yet.

I thought there was no interest... as you already used Vagrant VM to build it... If you think its useful, please merge it (or play with it and merge it)

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - .:/app
+    working_dir: /app/buildroot
+    command: bash -c "make BR2_EXTERNAL=../aa_wireless_dongle/ O=output/rpi0w raspberrypi0w_defconfig && cd output/rpi0w && make"
+
+  build_rpi3a:
+    build: .
+    volumes:
+      - .:/app
+    working_dir: /app/buildroot
+    command: bash -c "make BR2_EXTERNAL=../aa_wireless_dongle/ O=output/rpi3a raspberrypi3a_defconfig && cd output/rpi3a && make"
```
Comment (hkfuertes) [inline docker-compose.yml:26]: This might be removed until the other PR is merged. (https://github.com/nisargjhaveri/AAWirelessDongle/pull/11)

**Comment by nisargjhaveri** [comment]
Hi, sorry for the delay. I had played around with this, but was getting into some permission issues with the setup. Did play around some more and came up with 9d9dd9c27779c1f10969f5347e795bf0d65c3be8. Please have a look and suggest if it can be better? I'm new to docker, but seems much better than using Vagrant for this. Thanks for suggesting and getting me started on this!

**Comment by hkfuertes** [comment]
I have seen your `docker` branch... nice!!!
Please close this PR when you think it fits :)

Thank you!

**Comment by nisargjhaveri** [comment]
9d9dd9c27779c1f10969f5347e795bf0d65c3be8 adds support for building with docker. Thanks a lot for this. :)

**Pull request closed without merging**